benchmarks/src/jmh/kotlin/benchmarks/ChannelBenchmark.kt

```diff
@@ -0,0 +1,151 @@
+package benchmarks
+
+import kotlinx.coroutines.*
+import kotlinx.coroutines.channels.*
+import org.openjdk.jmh.annotations.*
+import java.util.concurrent.*
+
+@Warmup(iterations = 7, time = 1)
+@Measurement(iterations = 10, time = 1)
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.NANOSECONDS)
+@State(Scope.Benchmark)
+@Fork(1)
+open class ChannelBenchmark {
+    // max coroutines launched per benchmark
+    // to allow for true parallelism
+    val cores = 4
+
+    //                4 KB,   40 KB,   400 KB,      4 MB,      40 MB,      400 MB
+    @Param("1000", "10000", "100000", "1000000", "10000000", "100000000")
+    var count: Int = 0
+
+    // 1. Preallocate.
+    // 2. Different values to avoid helping the cache.
+    val maxCount = 100000000
+    val list = ArrayList<Int>(maxCount).apply {
+        repeat(maxCount) { add(it) }
+    }
+
+    @State(Scope.Benchmark)
+    open class UnlimitedChannelWrapper {
+        //                0,      4 MB,      40 MB,      400 MB
+        @Param("0", "1000000", "10000000", "100000000")
+        private var prefill = 0
+
+        lateinit var channel: Channel<Int>
+
+        val maxCount = 100000000
+        val list = ArrayList<Int>(maxCount).apply {
+            repeat(maxCount) { add(it) }
+        }
+
+        @Setup(Level.Invocation)
```

Comment:
Why it has to be done before every benchmark function invocation and not once per trial?

Reply:
It's a tradeoff. After each invocation, there could be a little extra items in the channel, which can accumulate with iterations. I can rewrite `runSendReceive` to leave channel with the same number of elements as it came in with, but then it will slightly affect the benchmark. Possibly negligible, since it's only up to 4 items each time...